experiment/selinux/src/selinux.rs

```diff
@@ -0,0 +1,433 @@
+use crate::xattr::*;
+use nix::unistd::gettid;
+use nix::sys::statfs;
+use nix::errno::Errno;
+use std::path::PathBuf;
+use std::fs::File;
+use std::io::{self, Read};
+use std::os::fd::{AsFd, AsRawFd};
+use std::sync::atomic::{AtomicBool, Ordering};
+
+const XATTR_NAME_SELINUX: &str = "security.selinux";
+const ERR_EMPTY_PATH: &str = "empty path";
+static HAVE_THREAD_SELF: AtomicBool = AtomicBool::new(false);
+static INIT_DONE: AtomicBool = AtomicBool::new(false);
```

Comment:
Can a structure or something have them to prevent from using `static`.

Reply:
Thanks. Changed the code using struct.